From 11c8b311004e14939cd741bc3d8bb28feb5bcdb1 Mon Sep 17 00:00:00 2001
From: "Liu, Xinwu" <xinwu.liu@intel.com>
Date: Wed, 26 Jun 2019 09:46:02 +0800
Subject: [PATCH] WA: forcing the preferences file name

forcing the name to com.android.camera2_preferences_camera_video0.xml.
logs:
java.lang.IllegalArgumentException:
File com.android.camera2_preferences_camera_/dev/video0.xml contains a path
separator

Change-Id: Icf0db44098add8f12110dd9c94d73180dcce1a83
Tracked-On:
Signed-off-by: Liu, Xinwu <xinwu.liu@intel.com>
---
 src/com/android/camera/settings/SettingsManager.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/com/android/camera/settings/SettingsManager.java b/src/com/android/camera/settings/SettingsManager.java
index c5c494ea8..65bf2074b 100644
--- a/src/com/android/camera/settings/SettingsManager.java
+++ b/src/com/android/camera/settings/SettingsManager.java
@@ -128,8 +128,10 @@ public class SettingsManager {
     protected SharedPreferences openPreferences(String scope) {
         synchronized (mLock) {
             SharedPreferences preferences;
+		//String name = mPackageName + newscope;
+		String name = "com.android.camera2_preferences_camera_video0";
             preferences = mContext.getSharedPreferences(
-                    mPackageName + scope, Context.MODE_PRIVATE);
+                    name, Context.MODE_PRIVATE);
 
             for (OnSharedPreferenceChangeListener listener : mSharedPreferenceListeners) {
                 preferences.registerOnSharedPreferenceChangeListener(listener);
-- 
2.21.0

